src/commands/rm.rs

```diff
@@ -0,0 +1,65 @@
+use crate::errors::ShellError;
+use crate::object::{Primitive, Value};
+use crate::prelude::*;
+use std::path::{Path, PathBuf};
+
+pub fn rm(args: CommandArgs) -> Result<OutputStream, ShellError> {
+    let cwd = args
+        .env
+        .lock()
+        .unwrap()
+        .path()
+        .to_path_buf();
+
+    if args.args.positional.is_none() {
+        return Err(ShellError::maybe_labeled_error(
+            "rm requires a filepath",
+            "can not delete",
+            args.name_span,
+        ));
+    }
+
+    let positional = match args.args.positional {
+        None => return Err(ShellError::string("rm requires a filepath")),
+        Some(p) => p,
+    };
+
+    let mut full_path = PathBuf::from(cwd);
+
+    match &(positional[0].item) {
+        Value::Primitive(Primitive::String(s)) => match s.as_str() {
+            "." | ".." => return Err(ShellError::labeled_error(
+                                  "\".\" and \"..\" may not be removed", 
+                                  "",
+                                  args.name_span.unwrap())),
+            _ => full_path.push(Path::new(s)),
+        },
+        _ => {}
+    }
+
+    let is_recursive = match positional.get(1) {
+        Some(Spanned {
+            item: Value::Primitive(Primitive::String(s)),
+            ..
+        }) if s == "--recursive" => true,
```

Comment:
There's a better way to do this now, if you're interested. You can see it with the config and open command. The named arguments are now separate from positional, and you can have them parsed for you ahead of time by setting them in the CommandConfig of the command.

Reply:
of course! Will investigate.